helix-view/src/clipboard.rs

```diff
@@ -37,18 +69,24 @@ pub fn get_clipboard_provider() -> Box<dyn ClipboardProvider> {
         command_provider! {
             paste => "wl-paste", "--no-newline";
             copy => "wl-copy", "--type", "text/plain";
+            primary_paste => "wl-paste", "-p", "--no-newline";
+            primary_copy => "wl-copy", "-p", "--type", "text/plain";
         }
     } else if env_var_is_set("DISPLAY") && exists("xclip") {
         command_provider! {
             paste => "xclip", "-o", "-selection", "clipboard";
             copy => "xclip", "-i", "-selection", "clipboard";
+            primary_paste => "xclip", "-o";
+            primary_copy => "xclip", "-i";
         }
     } else if env_var_is_set("DISPLAY") && exists("xsel") && is_exit_success("xsel", &["-o", "-b"])
     {
         // FIXME: check performance of is_exit_success
         command_provider! {
             paste => "xsel", "-o", "-b";
             copy => "xsel", "--nodetach", "-i", "-b";
+            primary_paste => "xsel", "-o";
+            primary_copy => "xsel", "-i";
```

Comment:
Is there a reason why `--nodetach` is not used for the primary one?

Reply:
I has a freeze with it (xsel stuck untill I killed it via system monitor). It works fine without `--nodetach`, not sure why it was necessary there. If I'm incorrect in removing that I can test it again and re-enable.

Reply:
Should be okay. I'll leave this thread open for other reviewers.

Reply:
@CBenoit I don't think all neovim's options are directly portable. I had issues with `wl-copy --foreground` for example since the command would never return and stall out: https://github.com/helix-editor/helix/commit/994ff4b2694044c60b69265cb176d0e5c608cf49#diff-68b83255fe92299d6a62dcca53412b8976d5eda55d3640b4f0dcec474249ba27